scripts/hash_file/index.ts

```diff
@@ -0,0 +1,54 @@
+import { createHash } from 'crypto';
+import { open, FileReadResult } from 'fs/promises';
+
+export async function hashFile(path: string): Promise<Buffer> {
+    return await hashFileByParts(path, 0);
+}
+
+async function hashFileByParts(
+    path: string,
+    position: number,
+    previousHash?: Buffer
+): Promise<Buffer> {
+    const { buffer, bytesRead } = await getBytesToHash(path, position);
+
+    if (bytesRead != 0) {
+        const newHash = hashChunkWith(buffer, previousHash);
+        return hashFileByParts(path, position + bytesRead, newHash);
+    } else {
+        // No more bytes to hash, set as final hash for this file
+        if (previousHash !== undefined) {
+            return previousHash;
+        } else {
+            // TODO if we add support for 0 byte files we might need something like this
+            // return hashChunkWith(Buffer.from([]));
+            throw new Error(`Error: No hash was found for ${path}`);
+        }
+    }
+}
+
+async function getBytesToHash(
+    path: string,
+    position: number
+): Promise<FileReadResult<Buffer>> {
+    const file = await open(path, 'r');
+
+    // Read the bytes
+    // TODO Before having the stable file storage hooked up 120 worked. For right now 60 seems to be working. I think we could do more but I want to get everything in place before spending a lot of time fine tuning it
+    // TODO it would be great to get the size of the chunks from the canister, then we wouldn't have to every update this
+    const limit = 60 * 1024 * 1024; // Must be the same as on the canister end or hashes will not match
+    const buffer = Buffer.alloc(limit); // Allocate a Buffer for reading
```

Comment:
check if this is mutated and change to let if it is

Reply:
done